### PR TITLE
[RISC-V] Fix building errors in emitriscv64.cpp on windows

### PR DESCRIPTION
**Pull request by Bajtazar**
Fixes `emitriscv64.cpp` building errors on windows

Part of #100099 , #84834
cc: @dotnet/samsung @jakobbotsch

### CODE REVIEW

**Review by clamp03 — APPROVED** [review]

**Comment by jakobbotsch** [comment]
Looks like there are Linux build failures

**Comment by clamp03** [comment]
@Bajtazar Could you please check coreclr tests on linux at least? Thank you.

File: src/coreclr/jit/emitriscv64.cpp [hunk 1]
```diff
@@ -2748,48 +2748,50 @@ ssize_t emitter::emitOutputInstrJumpDistance(const BYTE* src, const insGroup* ig
     return distVal;
 }
 
-static constexpr size_t NBitMask(uint8_t bits)
+template <uint8_t Bits>
+static inline constexpr typename std::conditional<(Bits > 32), size_t, unsigned>::type NBitMask() noexcept
 {
-    return (static_cast<size_t>(1) << bits) - 1;
+    return static_cast<typename std::conditional<(Bits > 32), size_t, unsigned>::type>(
+        (static_cast<size_t>(1) << Bits) - 1);
```
Comment (jakobbotsch) [inline src/coreclr/jit/emitriscv64.cpp]: Are there any uses that need a 64 bit mask? IMO just making it unconditionally return an `unsigned` would be significantly more readable than this.

Comment (Bajtazar) [reply src/coreclr/jit/emitriscv64.cpp]: I was thinking about future refactors where some longer masks could be useful, but you're right, for now the 32-bit version is sufficient enough

File: src/coreclr/jit/emitriscv64.cpp [hunk 2]
```diff
@@ -2748,48 +2748,48 @@ ssize_t emitter::emitOutputInstrJumpDistance(const BYTE* src, const insGroup* ig
     return distVal;
 }
 
-static constexpr size_t NBitMask(uint8_t bits)
+static inline constexpr unsigned WordMask(uint8_t bits) noexcept
```
Comment (jakobbotsch) [inline src/coreclr/jit/emitriscv64.cpp:2751]: Nit: `noexcept` is used nowhere in the JIT sources, so I wouldn't add it here either.

**Review by jakobbotsch — APPROVED** [review]
Thanks!

**Review by yurai007 — APPROVED** [review]

**Review by tomeksowi — APPROVED** [review]

**Review by sirntar — APPROVED** [review]

**Pull request merged**